cmd/root.go

```diff
@@ -66,6 +68,55 @@ func defaultConfig() string {
 	return filepath.Join(cfgDir, "bomctl", "bomctl.yaml")
 }
 
+func getDocumentCompletionArgs(
+	cmd *cobra.Command,
+	_ []string,
+	toComplete string,
+) ([]string, cobra.ShellCompDirective) {
+	backend, err := db.BackendFromContext(cmd.Context())
+	if err != nil {
+		logger.New("").Fatal(err)
+	}
+
+	comps := []string{}
+
+	defer backend.CloseClient()
+
+	documents, err := backend.GetDocumentsByIDOrAlias()
+	if err != nil {
+		backend.Logger.Fatal(err)
+	}
+
+	for _, document := range documents {
+		documentID := document.GetMetadata().GetId()
+		if slices.Contains(comps, documentID) {
+			continue
+		}
```

Comment:
We could save some complexity here by leveraging `sliceutil.Extract` or `sliceutil.Filter`